plugins/notifications-common/report.api.md

```diff
@@ -52,6 +53,9 @@ export type NotificationPayload = {
   topic?: string;
   scope?: string;
   icon?: string;
+  metadata?: {
+    [KMetadataKey in string]?: JsonValue;
```

Comment:
Related to the registry discussion above, `JsonValue` here is a tricky one. I do think it's what it should be, since only allowing strings is very limiting and leads to ugly workarounds. It does require some careful parsing though, and/or schemas. It could be that we for example want a JSON Schema associated with every well-known metadata key or something like that? DX there is somewhat important too with the best care being that there are TypeScript types as well.